From b06c89a4b491c3137da0325212fb7481ecabfe19 Mon Sep 17 00:00:00 2001
From: Daniel Petry <dpetry@anaconda.com>
Date: Tue, 21 Jan 2025 17:45:23 -0600
Subject: [PATCH 07/13] Add conda prefix to inductor include paths

Currently inductor doesn't look in conda's includes and libs. This results in
errors when it tries to compile, if system versions are being used of
dependencies (e.g., sleef).

Note that this is for inductor's JIT mode, not its AOT mode, for which the
end user provides a <filename>_compile_flags.json file.

Updated to use `sys.prefix` per
https://github.com/conda-forge/pytorch-cpu-feedstock/issues/424.
---
 torch/_inductor/cpp_builder.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/torch/_inductor/cpp_builder.py b/torch/_inductor/cpp_builder.py
index e2cb445ed10..120fae77475 100644
--- a/torch/_inductor/cpp_builder.py
+++ b/torch/_inductor/cpp_builder.py
@@ -1410,6 +1410,7 @@ def get_cpp_torch_options(
         + python_include_dirs
         + torch_include_dirs
         + omp_include_dir_paths
+        + [sys.prefix + '/include']
     )
     cflags = sys_libs_cflags + omp_cflags
     ldflags = omp_ldflags
